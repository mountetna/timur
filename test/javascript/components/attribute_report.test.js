import React from 'react';
import {Provider} from 'react-redux';
import {mockStore} from '../helpers';
import renderer from 'react-test-renderer';
import AttributeReport from '../../../lib/client/jsx/components/model_map/attribute_report';

const monster = require('../fixtures/template_monster.json');

describe('AttributeReport', () => {
  const store = mockStore({ });

  it('renders', () => {
    const tree = renderer
    .create(<Provider store={store}>
      <AttributeReport attribute={monster.attributes.name} />
      </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
