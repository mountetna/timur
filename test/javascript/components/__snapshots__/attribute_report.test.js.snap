// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeReport renders 1`] = `
<div
  className="MuiGrid-root makeStyles-attribute_report-2"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-attribute_card-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root makeStyles-heading-7 undefined MuiGrid-container"
    >
      <h6
        className="MuiTypography-root makeStyles-name-8 MuiTypography-h6"
      >
        Attribute
      </h6>
      Â 
      <h6
        className="MuiTypography-root makeStyles-title-9 MuiTypography-h6"
      >
        name
      </h6>
      <div
        className="MuiGrid-root MuiGrid-container"
        style={
          Object {
            "alignItems": "end",
            "flex": 1,
            "justifyContent": "right",
            "width": "auto",
          }
        }
      />
    </div>
    <div
      className="MuiCardContent-root makeStyles-content-4"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
        >
          attribute_type
        </div>
        <div
          className="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
        >
          identifier
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
        >
          display_name
        </div>
        <div
          className="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
        >
          Name
        </div>
      </div>
    </div>
  </div>
</div>
`;
